libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+import { Moment } from 'moment';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'tamu-gisc-time-map',
+  templateUrl: './time-map.component.html',
+  styleUrls: ['./time-map.component.scss']
+})
+export class TimeMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+
+  private maxDate: string;
+  private mortalButtonToggled: boolean;
+  private stateButtonToggle: boolean;
+  private selectedDate: string;
+  private stateData: object;
+  private countyData: object;
+
+  public ngOnInit() {
+    const initObject = this;
+
+    this.mortalButtonToggled = false;
+    this.stateButtonToggle = false;
+    this.reloadData(this.mortalButtonToggled);
+
+    this.mapService.loaded.subscribe((map) => {
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'covid-state',
+          type: 'fill',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          layout: {
+            // make layer visible by default (change to none or visible for debugging)
+            visibility: 'none'
+          }
+        },
+        'road-minor-low'
+      );
+
+      map.addLayer(
+        {
+          id: 'covid-county',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#ffffff'
+          }
+        },
+        'covid-state'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'covid-county', function(e) {
+        const feature = e.features[0];
+        const selectedCounty = initObject.countyData[initObject.selectedDate].filter(
+          (county) => county.fips === feature.properties.fips
+        );
+
+        document.getElementById('info-box').innerHTML =
```

Comment:
Do not use direct `document` methods to access DOM elements.

Prefer the use of the `@ViewChild` decorator using template references to query the DOM.

```html
<div #reference>
    <!-- Stuff goes here -->
</div>
```

```js
@ViewChild('reference') public container: ElementRef;

this.container.nativeElement // <-- This is your element.
```

Reply:
removed usage of direct document api access